### PR TITLE
Support commits for reusing common modules in other projects.

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

**Review by oleorhagen — COMMENTED** [review]
Last commit smells a bit to me.

Should we not now package it as an independent library, and use it. Instead of manipulating paths and adding new config files?

**Comment by kacf** [comment]
If we wanted to go the libntech route, then yes. But I don't think this is quite at that level yet, it has not been proven in any other project, and we are not aiming for binary compatibility or anything like that, at least not now. This is a pure vendor-style static dependency, so my plan was to use it exactly the same way that we're using it in the client.

Does it make sense? (And did I understand you correctly?)

File: main.cmake [hunk 1]
```diff
@@ -1,3 +1,6 @@
+set(MENDER_SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
+set(MENDER_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/src")
```
Comment (lluiscampos) [inline main.cmake:2]: `MENDER_SOURCE_DIR` and `MENDER_SRC_DIR` are very similar. Could we call the former one `MENDER_TOPLEVEL_DIR`, `MENDER_ROOT_DIR` or something similar?

**Review by lluiscampos — APPROVED** [review]

**Comment by kacf** [comment]
For the record, I misunderstood @oleorhagen, I thought he wanted a shared library, but he was talking about separate projects on the cmake level, which can then be imported.

@oleorhagen: Check it now. The split has been completely rewritten, and I think this turned out much nicer!

File: main.cmake (hunk 1 above)
Comment (kacf) [reply main.cmake]: With the changes after oleorhagen's comments, the variable was removed, so this is irrelevant now.

**Comment by kacf** [comment]
Only the two last commits are new, btw.

File: src/common/CMakeLists.txt [hunk 2]
```diff
@@ -156,3 +156,7 @@ if (MENDER_USE_SYSTEMD)
 else()
   target_sources(common_watchdog PRIVATE watchdog/noop/watchdog.cpp)
 endif()
+
+# Header-only.
+add_library(common_state_machine INTERFACE)
```
Comment (oleorhagen) [inline src/common/CMakeLists.txt:213]: Clever!

**Review by oleorhagen — APPROVED** [review]
Very nice!!

**Pull request merged**